### PR TITLE
add umami analytics to app

### PR DESCRIPTION
**Pull request by IgboPharaoh**
Adds umami analytics to app using #28 as a guideline

### CODE REVIEW

File: components/drawer.tsx [hunk 1]
```diff
@@ -87,6 +87,8 @@ export default function Drawer() {
                             className="font-semibold text-lg"
                             key={link.text}
                             href={link.linkTo}
+                            data-umami-event={`navigation-item-clicked`}
```
Comment (Extheoisah) [inline components/drawer.tsx:90]: Let's add the link text so its recognizable at a glance on umami
```suggestion
                            data-umami-event={`nav-${link.text}-clicked`}
```

File: components/nav-bar.tsx [hunk 2]
```diff
@@ -23,7 +23,12 @@ export default function NavBar() {
                             isMobile={false}
                         />
                         {links.map((link) => (
-                            <Link key={link.text} href={link.linkTo}>
+                            <Link
+                                key={link.text}
+                                href={link.linkTo}
+                                data-umami-event={`navigation-item-clicked`}
```
Comment (Extheoisah) [inline components/nav-bar.tsx:29]:
```suggestion
                                data-umami-event={`nav-${link.text}-clicked`}
```

File: app/career/page.tsx [hunk 3]
```diff
@@ -117,6 +117,8 @@ export default function Career() {
                                             href="https://spiral.xyz/#grants"
                                             target="_blank"
                                             rel="noopener noreferrer"
+                                            data-umami-event={`funding-org-clicked`}
```
Comment (Extheoisah) [inline app/career/page.tsx:120]: I don't see the event text specific to each link in the Umami dashboard. So I suggest we add the funding source's name to the event itself. The same applies to other funding sources
```suggestion
                                            data-umami-event={`funding-org-spiral-clicked`}
```

File: components/pointer.tsx [hunk 4]
```diff
@@ -20,6 +20,8 @@ export default function Pointer({
             <Button
                 href={`${jumpTo}`}
                 className={clsx("w-full px-3", className)}
+                data-umami-event={`hero-section-cta`}
```
Comment (Extheoisah) [inline components/pointer.tsx:23]:
```suggestion
                data-umami-event={`hero-section-cta-${btnText}`}
```

**Review by Extheoisah — CHANGES_REQUESTED** [review]
Looks good. Just a few nits.
We should also disable tracking when not in prod so that we don't have skewed data that is mixed. Here's a template you can modify for this
```js
export const disableUmamiTrack = () => {
    if (typeof window !== "undefined") {
        const disabled = localStorage.getItem("umami.disabled")
        if (process.env.VERCEL_ENV !== "production") {
            if (!disabled) {
                localStorage.setItem("umami.disabled", "1")
            }
        } else if (disabled) {
            localStorage.removeItem("umami.disabled")
        }
    }
}
```
See the bottom of the docs here https://umami.is/docs/track-events

File: app/layout.tsx
Comment (IgboPharaoh) [inline app/layout.tsx:1]: @Extheoisah For disabling tracking in other environments, I already used tracker configuration to set the data domains, I want it to get data from: https://umami.is/docs/tracker-configuration.

**Review by Extheoisah — APPROVED** [review]

**Comment by kouloumos** [comment]
> We should also disable tracking when not in prod so that we don't have skewed data that is mixed

I can't see this as part of this PR

**Comment by Extheoisah** [comment]
@kouloumos 

Here -> https://github.com/bitcoin-dev-project/bitcoin-dev-project/pull/64/files#diff-eca96d2c09f31517696a26e1d0be4070e1fbab02831481bed006e275741d030bR35

Used the domain to disable the tracking on any site other than bitcoindevs.xyz

**Comment by Extheoisah** [comment]
> Also, umami allows you to clear analytics data. i think we should do it before getting this to production in order for the analytics during testing to be removed.

Yeah. already prompted @adamjonas for this. Pls check [discord](https://discord.com/channels/1189471179967315968/1236986988750704651/1250323839271833652)

**Pull request merged**